src/quantem/imaging/lattice.py

```diff
@@ -0,0 +1,2961 @@
+    ) -> "Vector":
+        """
+        Measure the polarization of atoms at one site with respect to atoms at another site.
+        Polarization is computed as a fractional displacement (da, db) of each atom in the
+        'measure' site relative to the expected position inferred from the nearest atoms
+        in the 'reference' site and the current lattice vectors. The expected position is
+        the mean of neighbor positions shifted by the lattice vector transform of the
+        fractional index difference.
+
+        Parameters
+        ----------
+        measure_ind : int
+            Index of the site whose polarization is to be measured.
+            This corresponds to the index in `positions_frac` used in `add_atoms()`.
+        reference_ind : int
+            Index of the reference site used to calculate polarization.
+            This corresponds to the index in `positions_frac` used in `add_atoms()`.
+        reference_radius : float | None, default=None
+            If provided, neighbors are selected by radius search (in pixels) using a KD-tree.
+            Must be at least 1 pixel. If None, neighbors are selected by k-nearest search.
+        min_neighbours : int | None, default=2
+            Minimum number of nearest neighbors used to calculate polarization. Must be >= 2
+            when using k-nearest search (i.e., when `reference_radius` is None).
+        max_neighbours : int | None, default=None
+            Maximum number of nearest neighbors to use. Required when `reference_radius` is None.
+        plot_polarization_vectors : bool, default=False
+            If True, plots the polarization vectors using `self.plot_polarization_vectors(...)`.
+        **plot_kwargs
+            Additional keyword arguments forwarded to the plotting function.
+
+        Returns
+        -------
+        out : quantem.core.datastructures.vector.Vector
+            A Vector object containing the polarizations with:
+            - shape=(1,)
+            - fields=("x", "y", "a", "b", "da", "db")
+            - units=("px", "px", "ind", "ind", "ind", "ind")
+            Here, (x, y) are positions in pixels, (a, b) are fractional indices,
+            and (da, db) are fractional displacements (polarization).
+
+        Raises
+        ------
+        ValueError
+            - If the lattice vectors are singular (cannot invert).
+            - If neither `reference_radius` nor both `min_neighbours` and `max_neighbours` are specified.
+            - If `reference_radius` < 1.
+            - If radius-based search fails to find at least `min_neighbours` for any atom.
+            - If k-nearest search is used and `min_neighbours` or `max_neighbours` is missing.
+            - If k-nearest search is used with `min_neighbours` < 2 or `max_neighbours` < 2.
+            - If `min_neighbours` > `max_neighbours`.
+            - If no atoms have any neighbors identified (increase `reference_radius`).
+        DeprecationWarning
```

Comment:
How can this be `Deprecated` if this is the first time we've added this functonality?